utils/wasm-optimizer/src/optimize.rs

```diff
@@ -188,6 +182,8 @@ pub fn do_optimization<P: AsRef<Path>>(
         .arg("-mvp")
         .arg("--enable-sign-ext")
         .arg("--enable-mutable-globals")
+        .arg("--limit-segments")
+        .arg("--pass-arg=limit-segments@1024")
```

Comment:
Consider to move `DATA_SEGMENTS_AMOUNT_LIMIT` into `gear-core` and use it between crates so constant is always the same